src/bitmap/store.rs

```diff
@@ -136,6 +136,28 @@ impl Store {
         }
     }
 
+    ///
+    /// Pushes `index` in the store only if it is greater than the current maximum value.
+    /// It is up to the caller to have validated value > self.max()
+    ///
+    /// # Panics
+    ///
+    /// If debug_assertions enabled and value is > self.max()
+    pub(crate) fn push_unchecked(&mut self, index: u16) {
```

Comment:
Maybe we can simplify `push` here by just calling `push_unchecked` when we know that the push can be done safely?

Reply:
Can you maybe call `push_unchecked` from `push`, then?